src/lib/Permission/PermissionCheckerInterface.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzPlatformAdminUi\Permission;
+
+use eZ\Publish\API\Repository\Values\Content\Location;
+
+interface PermissionCheckerInterface
+{
+    /**
+     * @param $hasAccess
+     * @param string $class
+     *
+     * @return array
+     */
+    public function getRestrictions($hasAccess, string $class): array;
+
+    /**
+     * @param \eZ\Publish\API\Repository\Values\Content\Location $location
+     * @param array|bool $hasAccess
+     *
+     * @return bool
+     */
+    public function canCreateInLocation(Location $location, $hasAccess): bool;
```

Comment:
I'd like to hear others opinion on this but for me `Permission` namespace is very broad so `PermissionCheckerInterface` having `canCreateInLocation` method makes it badly designed. It should has more specific name i.e. `LocationBasedPermissionCheckerInterface` or the method should be a part of completely different interface.